lib/collection/src/collection/shard_transfer.rs

```diff
@@ -418,7 +420,24 @@ impl Collection {
             }
 
             if replica_set.is_dummy().await {
+                // We can reach here because of either of these:
+                // 1. Qdrant is in recovery mode (shard not dirty), and user intentionally triggered a transfer
+                // 2. Shard is dirty (shard initializing flag), and Qdrant automatically triggered a transfer to fix it (note: initializing flag means there must be another replica)
+                //
+                // In both cases, it's safe to drop existing local shard data
+                let was_dirty = replica_set.is_dirty().await;
+                log::debug!(
+                    "Initiating transfer to dummy shard {} with dirty = {}. Initializing empty local shard first",
+                    replica_set.shard_id,
+                    was_dirty
+                );
                 replica_set.init_empty_local_shard().await?;
+
+                if was_dirty {
+                    // TODO: Shouldn't we do this after we start the transfer?
+                    let shard_flag = shard_initializing_flag_path(&collection_path, shard_id);
+                    tokio::fs::remove_file(&shard_flag).await?;
+                }
```

Comment:
Let's remove the dirty flag and remove the initializing flag unconditionally (or more specifically, if it exists)